### PR TITLE
Update SA1302 code fix to support conflict resolution

### LINKED ISSUE #436: Update SA1302 (InterfaceNamesMustBeginWithI) to handle conflicting names
The following code results in an SA1302 warning. If the code fix is applied, it introduces a new compiler error.

```csharp
public interface Foo { }
public interface IFoo { }
public class Bar : Foo { }
```

We first need to decide how we want to handle this case (and similar), followed by implementation and tests.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #436

Edit: ~~Need to update this to ensure a nested type isn't renamed to the same name as the enclosing type.~~ This is now working.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs [hunk 1]
```diff
@@ -185,6 +185,160 @@ public interface FileOpenDialog111
...
+{
+    int IFoo { get; }
+}";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(2, 18);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedCode, cancellationToken: CancellationToken.None).ConfigureAwait(false);
+        }
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs]: :question:  Why is the interface renamed to `IFoo1` here? There clearly is no conflict.

**Comment by vweijsters** [comment]
:+1: Looks good to me, besides the question about a testcase.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs (hunk 1 above)
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs]: :arrow_right: It contains a member named `IFoo`. Members are not allowed to have the same name as the containing type, so either we don't use the name `IFoo` for the interface, or we have to rename both the interface and the property.

Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs]: Members can have the same name as the containing type, there is nothing preventing that. The compiler accepts it, and so does StyleCop Analyzers. Its not desirable, but that should the subject for a new diagnostic imho.

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs]: Looks like this is partially correct. If the containing type is a `class` or `struct`, you get CS0542. If the containing type is an `interface` or `enum`, you don't get an error.

**Pull request merged**